pkg/agent/core/state.go

```diff
@@ -667,8 +671,11 @@ func (s *state) desiredResourcesFromMetricsOrRequestedUpscaling(now time.Time) (
 	// 2. Cap the goal CU by min/max, etc
 	// 3. that's it!
 
+	hasMetrics := s.Metrics != nil && (!*s.scalingConfig().EnableLFCMetrics || s.LFCMetrics != nil)
```

Comment:
Maybe instead check separately for system metrics and lfc metrics, and if one exists, but the other doesn't, we just use the one which exists to compute maximum?

This can help in an event that e.g. LFC metric exposing/fetching breaks, and we would still be able to estimate goalCU based off CPU and memory.

Reply:
Yeah, this is a tricky situation... the thing I was trying to prevent was flushing caches on autoscaler-agent restart because goalCU was low before fetching LFC metrics.

I thought about potentially making making it a "min" goalCU if one of the metrics is missing (i.e., allow scaling _up_ to that amount, but don't scale down if we're already above it). But that's somewhat complex to implement in the current function.

Curious for your thoughts on this.

Reply:
Implemented this in 43a75dcd34103e3096c9bf722fe310cc3f10ea03.

Follow-up is in #1011.